valora-inc/wallet
# feat: add integration with supercharge v2 backend

### Description

This PR makes the wallet work with the v2 supercharge backend. Of note:
- add a new type for the v2 response
- add a new code path for the v2 stuff when fetching the rewards / displaying the rewards on the supercharge screen / claiming the rewards
- update unit tests

Not included in this PR: handling the error when the user is not verified - will make a follow up PR because this one is getting massive.

### Test plan

We should be able to:
- see and claim rewards as usual with the current system
- see and claim rewards in the new system
- see and claim rewards in the new system, when some of the rewards come from the old system

### Related issues

- Fixes RET-513

### Backwards compatibility

Y
